apps/cowswap-frontend/src/modules/injectedWidget/hooks/useAppCodeWithWidgetMetadata.ts

```diff
@@ -0,0 +1,46 @@
+import { useMemo } from 'react'
+
+import { environmentName } from '@cowprotocol/common-utils'
+
+import { AppDataWidget } from 'modules/appData/types'
+
+import { useInjectedWidgetMetaData } from './useInjectedWidgetMetaData'
+
+export interface AppCodeWithWidgetMetadata {
+  appCode: string
+  environment?: string
+  widget?: AppDataWidget
+}
+
+/**
+ *
+ * @param appCode of the app
+ *
+ * @returns the appCode that should be used in the metadata, and the widget metadata if it exists
+ */
+export function useAppCodeWithWidgetMetadata(appCode: string | null): AppCodeWithWidgetMetadata | null {
+  const injectedWidgetMetadata = useInjectedWidgetMetaData()
+
+  const appCodeWidget = injectedWidgetMetadata?.appCode
+
+  return useMemo(() => {
+    if (!appCode) {
+      return null
+    }
+
+    if (appCodeWidget) {
+      return {
+        appCode: appCodeWidget,
+        widget: {
+          appCode,
```

Comment:
If there's a widget appCode it's flipped with the app's appCode??

Reply:
Yes, this is the point. We add here the app (which is the host app). Our appCode is just the widget implementation

Reply:
Got it.
Can yo add a comment just so I don't forget down the line and switch it thinking it was a mistake 😅?

Reply:
https://github.com/cowprotocol/cowswap/pull/3371/commits/34efd471077c2acb899f4f7860c8f758bc3a6318